src/WeatherData.cpp

```diff
@@ -0,0 +1,139 @@
+#include "WeatherData.hpp"
+#include "merc.h"
+
+#include <fmt/format.h>
+
+using namespace fmt::literals;
+
+WeatherData weather_info;
+
+namespace {
+
+std::string_view to_string(WeatherData::Sky sky) {
+    switch (sky) {
+    case WeatherData::Sky::Cloudless: return "cloudless";
+    case WeatherData::Sky::Cloudy: return "cloudy";
+    case WeatherData::Sky::Raining: return "rainy";
+    case WeatherData::Sky::Lightning: return "lit by flashes of lightning";
+    }
+    return "not as expected";
+}
+
+std::string_view change_for(WeatherData::Sun sunlight) {
+    switch (sunlight) {
+    case WeatherData::Sun::Light: return "The day has begun.";
+    case WeatherData::Sun::Rise: return "The sun rises in the east.";
+    case WeatherData::Sun::Set: return "The sun slowly disappears in the west.";
+    case WeatherData::Sun::Dark: return "The night has begun.";
+    }
+    return "";
+}
+
+WeatherData::Sun sun_from_time(const TimeInfoData &tid) {
+    if (tid.hour() >= 5) {
+        if (tid.hour() == 5)
+            return WeatherData::Sun::Light;
+        if (tid.hour() < 19)
+            return WeatherData::Sun::Rise;
+        if (tid.hour() < 20)
+            return WeatherData::Sun::Set;
+    }
+    return WeatherData::Sun::Dark;
+}
+
+}
+
+int WeatherData::pressure_direction(const TimeInfoData &tid) const {
+    if (tid.is_summer())
+        return mmhg_ > 985 ? -2 : 2;
```

Comment:
I'd never noticed the diku/merc/rom folks had modeled air pressure.

Reply:
Was news to me too!